src/snowflake/cli/_plugins/object/manager.py

```diff
@@ -80,18 +80,35 @@ def create(self, object_type: str, object_data: Dict[str, Any]) -> str:
 
         try:
             response = rest.send_rest_request(url=url, method="post", data=object_data)
-            # workaround as SnowflakeRestful class ignores some errors, dropping their info and returns {} instead.
-            if not response:
-                raise ClickException(
-                    dedent(
-                        """                An unexpected error occurred while creating the object. Try again with --debug for more info.
-                Most probable reasons:
-                  * object you are trying to create already exists
-                  * role you are using do not have permissions to create this object"""
-                    )
-                )
             return response["status"]
         except BadRequest:
             raise ClickException(
-                "Incorrect object definition (arguments misspelled or malformatted)."
+                "400 bad request: Incorrect object definition (arguments misspelled or malformatted)."
             )
+        except HTTPError as err:
+            # according to https://docs.snowflake.com/developer-guide/snowflake-rest-api/reference/
+            match err_code := err.response.status_code:
+                case 401:
+                    raise ClickException(
+                        "401 unauthorized: role you are using does not have permissions to create this object."
+                    )
+                # error 403 should be handled by connector
+                # error 404 is handled by logic above
+                # error 405 should not happen under assumption that "all listable objects can be created"
+                case 408:
+                    raise ClickException(
+                        "408 timeout: the request timed out and was not completed by the server."
+                    )
+                case 409:
+                    raise ClickException(
+                        "409 conflict: object you're trying to create already exists."
+                    )
+                # error 410 is a network maintenance debugging - should not happen to the user
+                case 429:
+                    raise ClickException(
+                        "429 too many requests. The number of requests hit the rate limit."
+                    )
+                case 500 | 503 | 504:
+                    raise ClickException(f"{err_code} internal server error.")
+                case _:
+                    raise
```

Comment:
Maybe extract handling to function? WDYT?

Reply:
It actually looks better. Fixed